builtin/providers/azurerm/resource_arm_managed_image.go

```diff
@@ -0,0 +1,421 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmManagedImage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedImageCreate,
+		Read:   resourceArmManagedImageRead,
+		Update: resourceArmManagedImageCreate,
+		Delete: resourceArmManagedImageDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"source_virtual_machine_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"os_disk.os_disk_ostype"},
+			},
+
+			"os_disk": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"os_disk_ostype": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.Linux),
+								string(compute.Windows),
+							}, true),
+						},
+
+						"os_disk_osstate": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.Generalized),
+								string(compute.Specialized),
```

Comment:
Are there any times when "Specialized" is actually valid?  I don't see any here: https://docs.microsoft.com/en-us/rest/api/manageddisks/images/images-create